### DISCUSSION TITLE
How to set up the flow calculation area?

### DISCUSSION THREAD

**Discussion by zevenlordsa** [NatLabRockies/floris#716] [Q&A] [answered]
"WARNING The calculated flow field contains points outside of the the user-defined heterogeneous inflow bounds. For these points, the interpolated value has been filled with the freestream wind speed. If this is not the desired behavior, the user will need to expand the heterogeneous inflow bounds to fully cover the calculated flow field area."

When I use the heterogeneous inflow config that set by myself, there is a warning coming out. So how can I set the calculated area to fit it?
 Here is the CSV file contains my config，the file contains more than 31w data points, processed from wind field files generated by Sowfa.
[Low.csv](https://github.com/NREL/floris/files/12854316/Low.csv)
.

**Comment by rafmudaf** [comment]
Hi @zevenlordsa please share the `heterogenous_inflow_config` block of your input file.

**Reply by zevenlordsa** [reply]
Thank you sir, I don't know how to add an attachment in github, so I sent it to your email.
Have a great day sir!

**Reply by zevenlordsa** [reply]
```python
data = pd.read_csv('desktop/wind.csv')
speed_multipliers = [data['speed_multipliers'].values.tolist()]
x_locs = data['x_locs'].values.tolist()
y_locs = data['y_locs'].values.tolist()
z_locs = data['z_locs'].values.tolist()

heterogenous_inflow_config = {
    'speed_multipliers': speed_multipliers,
    'x': x_locs,
    'y': y_locs,
    'z': z_locs,
}
```

**Reply by rafmudaf** [reply]
> Thank you sir, I don't know how to add an attachment in github, so I sent it to your email.

@zevenlordsa thanks for sending the csv file via email, but it is more effective to share all information on GitHub for the benefit of the FLORIS community. You can drag the file to the text entry field directly on the GitHub webpage to attach it. Please edit your post above with the heterogeneous configuration (click the three dots on the top right of the post and select "Edit") to also include the csv file.

**Reply by zevenlordsa** [reply]
OK sir, I've attached the csv file above, and here is my full script. Thank  you for your time!
```python
import matplotlib.pyplot as plt
import numpy as np
from floris.tools import FlorisInterface
from floris.tools.visualization import visualize_cut_plane
import pandas as pd

data = pd.read_csv('Low.csv')
speed_multipliers = [data['speed_multipliers'].values.tolist()]
x_locs = data['x_locs'].values.tolist()
y_locs = data['y_locs'].values.tolist()
z_locs = data['z_locs'].values.tolist()

heterogenous_inflow_config = {
    'speed_multipliers': speed_multipliers,
    'x': x_locs,
    'y': y_locs,
    'z': z_locs,
}

D = 126
fi_3d = FlorisInterface("gch.yaml")
fi_3d.reinitialize(
    layout_x=[6 * D, 8 * D, 10 * D],
    layout_y=[1000, 1000, 1000],
)

fi_3d.reinitialize(heterogenous_inflow_config=heterogenous_inflow_config)
fi_3d.reinitialize(wind_shear=0.0)

horizontal_plane_3d = fi_3d.calculate_horizontal_plane(
    x_resolution=200,
    y_resolution=100,
    height=90.0,
)
# Create the plots
fig, ax_list = plt.subplots(1, 1, figsize=(10, 8))
visualize_cut_plane(
    horizontal_plane_3d, 
    ax=ax_list, 
    title="Horizontal"
)
plt.show()
```

**Comment by bayc** [comment, marked as answer]
@zevenlordsa Thanks for reaching out. The warning that you received is meant to alert the user in case there are points outside of their defined heterogenous inputs. In this case, this is occurring because of the default bounds that are used in FLORIS' `calculate_horizontal_plane` are beyond the bounds that you have defined in your `.csv` file. If this behavior is acceptable, you don't need to change annything and keep the code as is. If not, a simple fix would be to supply bounds to the `calculate_horizontal_plane` that are within your defined area using `x_bounds` and `y_bounds`, such as below:

```
horizontal_plane_3d = fi_3d.calculate_horizontal_plane(
    x_resolution=200,
    y_resolution=100,
    height=90.0,
    x_bounds=(626., 2004.),
    y_bounds=(626., 1374.),
)
```

Using these bounds and running your code with a smaller set of heterogenous data points (that still include the min/max x and y coordinates but use a subset to allow for faster calculation), I am able to get the figure below without the warning. I've included my down-selected set of heterogenous points for reference.

[low_downselect.csv](https://github.com/NREL/floris/files/12885073/low_downselect.csv)

![image](https://github.com/NREL/floris/assets/12664940/818af393-24a8-4fd7-85bb-50c4331ed370)

**Reply by rafmudaf** [reply]
Great description @bayc. I hope this resolves @zevenlordsa's questions, and it should also be included somewhere in the documentation. A good place for this might be as a section in [Background and Concepts](https://nrel.github.io/floris/floris_101.html) guide. I've opened #717 for this.

**Reply by zevenlordsa** [reply]
Wow thank you so much! I've tried using this method and it works!
